internal/k8s/vald/benchmark/scenario/scenario.go

```diff
@@ -0,0 +1,128 @@
+//
+// Copyright (C) 2019-2023 vdaas.org vald team <vald@vdaas.org>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package scenario
+
+import (
+	"context"
+	"time"
+
+	"github.com/vdaas/vald/internal/k8s"
+	v1 "github.com/vdaas/vald/internal/k8s/vald/benchmark/api/v1"
+	"github.com/vdaas/vald/internal/log"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type BenchmarkScenarioWatcher k8s.ResourceController
+
+type reconciler struct {
+	mgr         manager.Manager
+	name        string
+	namespaces  []string
+	onError     func(err error)
+	onReconcile func(ctx context.Context, operatorList map[string]v1.ValdBenchmarkScenario)
+	lopts       []client.ListOption
+}
+
+func New(opts ...Option) (BenchmarkScenarioWatcher, error) {
+	r := new(reconciler)
+	for _, opt := range append(defaultOpts, opts...) {
+		// TODO: impl error handling after implement functional option
```

Comment:
Is it better to sove TODO comment? 🤔

Reply:
fixed: https://github.com/vdaas/vald/pull/2216/commits/85b9e420be22649661a6ff3b9423b3f721687bac